autorest/version.go

```diff
@@ -12,7 +12,12 @@ const (
 	semVerFormat = "%s.%s.%s%s"
 )
 
+var version string
+
 // Version returns the semantic version (see http://semver.org).
 func Version() string {
-	return fmt.Sprintf(semVerFormat, major, minor, patch, tag)
+	if "" == version {
```

Comment:
I don't think this style is necessary in Go as you can't have an assignment here.